harness/determined/keras/_tf_keras_trial.py

```diff
@@ -401,56 +468,68 @@ def _load(self) -> None:
         self.model.load_weights(str(full_ckpt_path))
         load_optimizer_weights(self.model, full_ckpt_path)
 
-    def _save_checkpoint(self, path: pathlib.Path) -> workload.Response:
-        # We assume that at least one training step has completed when saving a
-        # checkpoint.
+        # Load callbacks.
+        cb_state_path = self.load_path.joinpath("determined-callbacks.v1.pkl")
+        if cb_state_path.exists():
+            with cb_state_path.open("rb") as f:
+                cb_state = pickle.load(f)
+        self.multiplexer._load_state(cb_state)
 
-        if not self.is_chief:
-            return workload.Skipped()
+    def run(self) -> None:
+        try:
+            self._launch_fit()
+        except det.errors.WorkerFinishedGracefully:
+            pass
 
-        # Save training data iterator position.
-        path.mkdir(parents=True, exist_ok=True)
+    def _launch_fit(self) -> None:
+        (
+            training_input,
+            batches_per_epoch,
+        ) = self._train_input_manager.get_training_input_and_batches_per_epoch()
 
-        # Save model.
-        self.model.save(path.joinpath("determined-keras-model.h5"), save_format="h5")
+        self.model.fit(
+            training_input,
+            callbacks=self.callback_list,
+            shuffle=False,
+            steps_per_epoch=batches_per_epoch,
```

Comment:
We previously talked about making this a very large number to get the expected metric performance.